ogre2/src/Ogre2GpuRays.cc

```diff
@@ -757,30 +757,59 @@ void Ogre2GpuRays::UpdateSampleTexture()
     return points;
   };
 
-  // Real Lidar has a polling rate of 240k pps
-  // We keep time in sync with real lidar,
-  // but keep polling rate constant according to image buffer size
-  double time = this->dataPtr->timer.ElapsedTime().count();
-  double span = time - this->dataPtr->lastElapsed;
-  this->dataPtr->lastElapsed = time;
-  
-  auto points = getLidarPoints(time, span, this->dataPtr->w2nd * this->dataPtr->h2nd);
-
   double hmin = this->AngleMin().Radian();
   double hmax = this->AngleMax().Radian();
   double vmin = this->VerticalAngleMin().Radian();
   double vmax = this->VerticalAngleMax().Radian();
   float *pDest = this->dataPtr->textureData;
 
+  std::vector<std::tuple<float, float>> points;
+  auto hdiff = hmax - hmin;
+  auto vdiff = vmax - vmin;
+
+  if (this->pattern == ScanningPattern::AVIA)
+  {
+    // Real Lidar has a polling rate of 240k pps
+    // We keep time in sync with real lidar,
+    // but keep polling rate constant according to image buffer size
+    double time = this->dataPtr->timer.ElapsedTime().count();
+    double span = time - this->dataPtr->lastElapsed;
+    this->dataPtr->lastElapsed = time;
+
+    points = getLidarPoints(time, span, this->dataPtr->w2nd * this->dataPtr->h2nd);
+
+    hdiff /= 2;
+    vdiff /= 2;
+  }
+
+  else if (this->pattern == ScanningPattern::RASTERIZATION)
+  {
+    hdiff /= this->dataPtr->h2nd;
+    vdiff /= this->dataPtr->w2nd;
+  }
+
   for (unsigned i = 0; i < this->dataPtr->h2nd; ++i)
   {
     for (unsigned j = 0; j < this->dataPtr->w2nd; ++j)
     {
       unsigned index = i * this->dataPtr->w2nd + j;
 
-      auto [x, y] = points[index];
-      x = hmin + (x + 1) / 2 * (hmax - hmin);
-      y = vmin + (y + 1) / 2 * (vmax - vmin);
+      float x, y, dx, dy;
```

Comment:
Let's initialize the variables, e.g.
```suggestion
      float x = 0.0f, y = 0.0f, dx = 0.0f, dy = 0.0f;
```

Reply:
Modified in the next commit.